### PR TITLE
[WIP] adding basic http authentication capabilities

### PR DESCRIPTION
**Pull request by Caroga**
**Work in progress**, not done yet. Need to test this fully yet. 
Any preliminary feedback would be appreciated. 

Contains the following: 
- if/else check to enable http authentication sequence.
- Check if the needed environment variables are set.
- Generates htpasswd file
- Enables configuration inside nginx
- Test nginx config modification

Todo:
- Add docker secrets compatibility: using docker swarm you can mount secrets on `/run/secrets/<secret_name>`, this way you can add the password without having it on your filesystem.

### CODE REVIEW

**Comment by Caroga** [comment]
When merged, this solves #70

**Comment by xirixiz** [comment]
Hi Caroga, thanks for the effort! I`ll test the Docker image Tonight with the changes you've made. Good point about Docker secrets.

**Comment by xirixiz** [comment]
Which images do you use yourself btw, arm or amd? I just created a tag: test-latest which is amd64 based. I`ll verify the changes and then merge your pull request.

I really should do some automated testing in the future 😃

**Comment by Caroga** [comment]
I've based this on amd64. I only have not tested it thoroughly through yet.
Will take a crack at it (and the docker secrets support) in about an hour
or 2.

Op do 10 okt. 2019 10:39 schreef Bram van Dartel <notifications@github.com>:

File: app/run.sh [hunk 1]
```diff
@@ -90,6 +90,39 @@ else:
...
+      if [[ -z "${HTTP_AUTH_PASSWORD}" ]]; then
+        _warn "Please provide a HTTP_AUTH_PASSWORD"
+        canWeContinue=false
+      fi
+      if [[ "${canWeContinue}" = false ]] ; then
+        _error "Cannot generate a valid .htpasswd file, please check above warnings."
+        exit 1
+      fi
+      _info "Generating htpasswd..."
+      printf ${HTTP_AUTH_USERNAME}":$(openssl passwd -apr1 "${HTTP_AUTH_PASSWORD}")\n" > /etc/nginx/htpasswd
```
Comment (xirixiz) [inline app/run.sh]: openssl package is not included in the Dockerfile

**Pull request merged**